### PR TITLE
Add responses processing in the Previewer controller

### PR DESCRIPTION
**Pull request by jsconan**
Related to: https://oat-sa.atlassian.net/browse/TAO-8191

Add the responses processing in the Previewer controller.
The code has been take from [`taoQtiItem`](https://github.com/oat-sa/extension-tao-itemqti/blob/bee471b6ba1d82bb2b1387f00607bf06119a4e37/controller/QtiPreview.php#L66), and it should probably require some tuning and refining.

### CODE REVIEW

File: controller/Previewer.php [hunk 1]
```diff
@@ -260,4 +287,98 @@ public function submitItem()
...
+                }
+                else {
+                    $variables[] = $var;
+                }
+            }
+            catch (OutOfRangeException $e) {
+                // A variable value could not be converted, ignore it.
+                // Developer's note: QTI Pairs with a single identifier (missing second identifier of the pair) are transmitted as an array of length 1,
+                // this might cause problem. Such "broken" pairs are simply ignored.
+                common_Logger::d("Client-side value for variable '${id}' is ignored due to data malformation.");
```
Comment (k-v-vladimir) [inline controller/Previewer.php]: It is better to use \oat\oatbox\log\LoggerAwareTrait

File: controller/Previewer.php [hunk 2]
```diff
@@ -260,4 +287,98 @@ public function submitItem()
...
+            }
+            catch (OutOfBoundsException $e) {
+                // No such identifier found in item.
+                common_Logger::d("The variable with identifier '${id}' is not declared in the item definition.");
```
Comment (k-v-vladimir) [inline controller/Previewer.php]: I would use \oat\oatbox\log\LoggerAwareTrait

File: controller/Previewer.php [hunk 3]
```diff
@@ -260,4 +287,98 @@ public function submitItem()
...
+            // Return the item session state to the client-side.
+            return [
+                'success' => true,
+                'displayFeedback' => true,
+                'itemSession' => self::buildOutcomeResponse($itemSession)
+            ];
+        }
+        catch(AssessmentItemSessionException $e) {
+            $msg = "An error occurred while processing the responses.";
+            throw new \RuntimeException($msg, 0, $e);
```
Comment (k-v-vladimir) [inline controller/Previewer.php]: import in "use" clause

File: controller/Previewer.php [hunk 4]
```diff
@@ -260,4 +287,98 @@ public function submitItem()
...
+        }
+        catch(taoQtiCommon_helpers_ResultTransmissionException $e) {
+            $msg = "An error occurred while transmitting a result to the target Result Server.";
+            throw new \RuntimeException($msg, 0, $e);
```
Comment (k-v-vladimir) [inline controller/Previewer.php]: import in "use" clause

**Comment by oatymart** [comment]
This branch is in active use and should be preserved until #21 is merged to develop.

**Pull request merged**